plot-stdin.py

```diff
@@ -0,0 +1,170 @@
+    """
+
+    ENABLE_ALTERNATE_SCREEN = b"\x1b[?1049h"
+    DISABLE_ALTERNATE_SCREEN = b"\x1b[?1049l"
+    MOVE_CURSOR_TO_ORIGIN = b"\x1b[0;0H"
+    HIDE_CURSOR = b"\x1b[?25l"
+    SHOW_CURSOR = b"\x1b[?25h"
+
+
+def ansi_scoped_enable(enable: bytes, disable: bytes):
+    sys.stdout.buffer.write(enable)
+    sys.stdout.flush()
+    try:
+        yield
+    finally:
+        sys.stdout.buffer.write(disable)
+        sys.stdout.flush()
+
+
+@contextlib.contextmanager
+def hide_cursor():
+    yield from ansi_scoped_enable(Escapes.HIDE_CURSOR, Escapes.SHOW_CURSOR)
+
+
+@contextlib.contextmanager
+def use_alternate_screen():
+    """
+    Switches to terminal's "alternate screen", clearing the plot from terminal on exit.
+    """
+    yield from ansi_scoped_enable(
+        Escapes.ENABLE_ALTERNATE_SCREEN,
+        Escapes.DISABLE_ALTERNATE_SCREEN,
+    )
+
+
+class Plot:
+    def __init__(self, series: typing.List[float]):
+        self.term_rows, self.term_cols = termios.tcgetwinsize(sys.stdout)
+        self.series = series
+        y_tics = [
+            min(series)
+            + (max(series) - min(series))
+            * (idx / (self.term_rows - self.x_axis_margin - 1))
+            for idx in range(self.term_rows - self.x_axis_margin)
+        ]
+        self.y_tics = list(reversed(["{:.02f}".format(tic) for tic in y_tics]))
+        # -1 for the axis lines
+        self.canvas = BrailleCanvas(
+            char_rows=self.term_rows - self.x_axis_margin - 1,
+            char_columns=self.term_cols - self.y_axis_margin - 1,
+        )
+
+        # Rectangles representing the value (src) and canvas (dst) coordinate space
+        # Used to convert incoming value series into something canvas will understand
+        src_rect = Rect(
+            top_left=Point(x=float(0), y=max(series)),
+            bottom_right=Point(x=float(len(series)), y=min(series)),
+        )
+        dst_rect = self.canvas.rect()
+
+        if src_rect.width == 0 or src_rect.height == 0:
+            return
+
+        def convert(point: Point):
+            mapped = map_coords(point, src_rect, dst_rect)
+            # BrailleCanvas expects y,x order
+            return (mapped.y, mapped.x)
+
+        points = [convert(Point(idx, val)) for idx, val in enumerate(series)]
+
+        for u, v in zip(points[:-1], points[1:]):
+            self.canvas.draw_line(u, v)
+
+    @property
+    def x_axis_margin(self) -> int:
+        """
+        Number of terminal columns reserved for y axis labels,
+        Does not include space use by the axis itself.
+        """
+        return 1
+
+    @property
+    def y_axis_margin(self) -> int:
+        """
+        Number of terminal columns reserved for y axis labels.
+        Does not include space use by the axis itself.
+        """
+        if self.y_tics:
+            return max(len(tic) for tic in self.y_tics) + 1
+        else:
+            return 0
+
+    @property
+    def x_axis(self) -> typing.List[str]:
```

Comment:
I don't understand what exactly this method represents - Please add a docstring.